src/ol_infrastructure/applications/micromasters/__main__.py

```diff
@@ -0,0 +1,174 @@
+from ol_infrastructure.lib.pulumi_helper import parse_stack
+from ol_infrastructure.lib.stack_defaults import defaults
+from ol_infrastructure.lib.vault import setup_vault_provider
+
+setup_vault_provider()
+micromasters_config = Config("micromasters")
+stack_info = parse_stack()
+network_stack = StackReference(f"infrastructure.aws.network.{stack_info.name}")
+micromasters_vpc = network_stack.require_output("applications_vpc")
+operations_vpc = network_stack.require_output("operations_vpc")
+micromasters_environment = f"micromasters-{stack_info.env_suffix}"
+aws_config = AWSBase(
+    tags={
+        "OU": "micromasters",
+        "Environment": micromasters_environment,
+        "Application": "micromasters",
+    }
+)
+
+# Create S3 bucket
+
+# Bucket used to store files from MicroMasters app.
+micromasters_bucket_name = f"ol-micromasters-app-{stack_info.env_suffix}"
+micromasters_bucket = s3.Bucket(
+    f"micromasters-{stack_info.env_suffix}",
+    bucket=micromasters_bucket_name,
+    versioning=s3.BucketVersioningArgs(
+        enabled=True,
+    ),
+    tags=aws_config.tags,
+    acl="private",
+    policy=json.dumps(
+        {
+            "Version": "2012-10-17",
+            "Statement": [
+                {
+                    "Effect": "Allow",
+                    "Action": [
+                        "s3:GetObject",
+                        "s3:ListAllMyBuckets",
+                        "s3:ListBucket",
+                        "s3:ListObjects",
+                        "s3:PutObject",
+                        "s3:DeleteObject",
+                    ],
+                    "Resource": [f"arn:aws:s3:::{micromasters_bucket_name}/*"],
+                }
+            ],
+        }
+    ),
+    cors_rules=[{"allowedMethods": ["GET", "HEAD"], "allowedOrigins": ["*"]}],
+)
+
+
+micromasters_iam_policy = iam.Policy(
+    f"micromasters-{stack_info.env_suffix}-policy",
+    description="AWS access controls for the MicroMasters application in the "
+    f"{stack_info.name} environment",
+    path=f"/ol-applications/micromasters/{stack_info.env_suffix}/",
+    name_prefix=f"micromasters-{stack_info.env_suffix}-application-policy-",
+    policy=lint_iam_policy(
+        {
+            "Version": "2012-10-17",
+            "Statement": [
+                {
+                    "Effect": "Allow",
+                    "Action": "s3:ListAllMyBuckets",
+                    "Resource": "*",
+                },
+                {
+                    "Effect": "Allow",
+                    "Action": [
+                        "s3:ListBucket*",
+                        "s3:PutObject",
+                        "s3:PutObjectAcl",
+                        "s3:GetObject*",
+                        "s3:DeleteObject*",
+                    ],
+                    "Resource": [
+                        f"arn:aws:s3:::{micromasters_bucket_name}",
+                        f"arn:aws:s3:::{micromasters_bucket_name}/*",
```

Comment:
Looking at https://github.com/mitodl/salt-ops/blob/ee5e2ccae3c85f13740d4b181bcd39e27a5707e0/pillar/heroku/micromasters.sls#L183-L184 it seems like we actually want to include that other bucket in this policy so we don't have to have two separate credentials.

Reply:
It looks like we're not handling that audit bucket at all in Pulumi. Should we follow suit with the others and transfer it to a new one that follows our new naming conventions as per Sar?

Reply:
Don't think it's worth messing with that audit bucket and I'll change pillar to use the existing creds as all the `EXAMS` pillar doesn't appear to be in use and the code will eventually be removed from the app. https://github.com/mitodl/micromasters/issues/5314